docs/modules/ROOT/pages/fungibleToken.adoc

```diff
@@ -0,0 +1,145 @@
+
+**Features and specifications NOT supported**
+
+- **Events** - Midnight does not currently support events, but this is planned on being supported in the future.
+- **Uint256 type** - There's ongoing research on ways to support uint256 in the future.
+- **Interface** - Compact currently does not have a way to define a contract interface.
+This library offers modules of contracts with free floating circuits;
+nevertheless, there are no means of enforcing that all circuits are provided.
+
+== Contract-to-contract calls
+
+Contract-to-contract calls are currently not supported in the Compact language.
+Due to this limitation, the current iteration of FungibleToken disallows transfers and mints to the `ContractAddress` type.
+Transferring tokens to a contract may result in those tokens being locked forever.
+The FungibleToken module, however, does provide `unsafe` circuit variants for users who wish to experiment with sending tokens to contracts.
+
+WARNING: The `unsafe` circuits will eventually be deprecated after Compact supports contract-to-contract calls—meaning
+`transfer`, `_mint`, etc. are planned to eventually allow the recipients to be of the `ContractAddress` type.
+
+== Usage
+
+:extensibility-pattern: xref:extensibility.adoc#the_module_contract_pattern[Module/Contract Pattern]
+:fungible-mint: xref:/api/fungibleToken.adoc#FungibleTokenModule-_mint[_mint]
+
+Import the FungibleToken module into the implementing contract.
+It's recommended to prefix the module with `FungibleToken_` to avoid circuit signature clashes.
+
+```typescript
+pragma language_version >= 0.15.0;
+
+import CompactStandardLibrary;
+import "./node_modules/@openzeppelin-compact/tokens/FungibleToken" prefix FungibleToken_;
+
+constructor(
+  name: Opaque<"string">,
+  symbol: Opaque<"string">,
+  decimals:Uint<8>
+) {
+  FungibleToken_initialize(name, symbol, decimals);
+}
+```
+
+Next, expose the ciruits that users may call in the contract.
+This library enables extensibility by following the rules of the {extensibility-pattern}.
+Note that circuits with a preceding underscore (`_likeThis`) are meant to be building blocks for implementing contracts.
+Exposing {fungible-mint} without some sort of access control, for example, would allow ANYONE to mint tokens.
+
+```typescript
+export circuit name(): Opaque<"string"> {
+  return FungibleToken_name();
+}
+
+export circuit symbol(): Opaque<"string"> {
+  return FungibleToken_symbol();
+}
+
+export circuit decimals(): Uint<8> {
+  return FungibleToken_decimals();
+}
+
+(...)
+```
+
+The following example is a simple token contract with a fixed supply that's minted to the passed recipient upon construction.
+
+```typescript
+// FungibleTokenFixedSupply.compact
+
+pragma language_version >= 0.15.0;
+
+import CompactStandardLibrary;
+import "./node_modules/@openzeppelin-compact/tokens/FungibleToken" prefix FungibleToken_;
```

Comment:
Great example! In the future we should add a non-trivial function to demonstrate proper usage of internal functions in the entrypoint contract.

Reply:
Thank you! And for sure! When we get Ownable up, we'll have a lot more flexibility